Governance Finalization (#51)
conda-incubator/governance: README.md

# Conda & Conda-Incubator Governance

This document outlines the policies and procedures that manage the conda community. This document recognizes that, while packaging and package tooling is deeply important to all of us, participation in the organization is voluntary. Individual members of the Steering Council, any sub-team, and external contributors choose to participate and may choose to leave at any time for any reason or for no reason.

Larger organizations, such as for-profit businesses, that contribute code and support to the conda organization are welcome to (but not required to) [maintain creative control](#federated-projects) of their contributions as long as such contributions adhere to the rules in this document. Such organizations are also welcome to [cede creative & maintenance control](#community-projects) of any specific project to the community as well.

We deeply appreciate all good faith contributions.

## Code of Conduct

The conda Organization's [Code of Conduct](CODE_OF_CONDUCT.md) governs all interactions within the conda Organization.  Please take a few minutes to familiarize yourself with it.

## Naming & Manifest

> This document defines the conda Organization as the body responsible for governing the tools, specifications, and libraries within the conda ecosystem, broadly speaking. This encompasses specific GitHub organizations, such as [`conda`](https://github.com/conda/) and [`conda-incubator`](https://github.com/conda-incubator/). Authors of particular software projects, such as [`conda`](https://github.com/conda/conda) or [`mamba`](https://github.com/mamba-org/mamba), are free to choose if and when their projects are included as part of the conda Organization.

Reviewer: ```suggestion
This document defines the conda Organization as the body responsible for governing the tools, specifications, and libraries within the conda ecosystem, broadly speaking. This encompasses specific GitHub organizations, such as [`conda`](https://github.com/conda/) and [`conda-incubator`](https://github.com/conda-incubator/). Authors of particular software projects, such as [`conda`](https://github.com/conda/conda) or [`mamba`](https://github.com/mamba-org/mamba), are free to choose when their projects are included as part of the conda Organization.
```

## Terminology

1. *conda* or *the conda Organization* : The conda Organization defined here, including `conda` and `conda-incubator`.
2. *The [conda project](https://github.com/conda/conda)* : The actual conda software itself, as distinct from the conda Organization.
3. *The conda ecosystem* : The broadest set of tools, packages, and people that use any conda related tooling. The conda Organization is a subset of *the conda ecosystem*. Members of the ecosystem should not feel compelled or obligated to participate in the conda Organization, though positive engagement is welcome.

## Teams & Roles

The primary teams participating in the conda Organization's activities are:

### Steering Council

The Steering Council is the governing body over the entire conda Organization. Members of the Steering Council have full voting rights within the organization. Steering Council members are the face of the organization, and are responsible for officially interfacing with external communities, organizations, non-profits, and companies. The Steering Council may create new sub-teams, as appropriate. Each Steering Council member is entitled to one vote on all elected matters.

### Project Teams

> Project Teams are responsible for the management of a software project consisting of one or more repositories and any software releases. Project Team members have the ability to merge pull requests into the repositories of the software project they maintain and produce releases of that software project. See the [Project Teams Details](#project-teams-details) section for more.

Reviewer: ```suggestion
Project Teams are responsible for the management of a software project consisting of one or more repositories and any software releases. Project Team members have the ability to merge pull requests into the repositories of the software project they maintain, produce releases of that software project, and all other duties related to repo maintenance. See the [Project Teams Details](#project-teams-details) section for more.
```

## Shared Funding Membership restrictions

**No more than 2 council members can have the same funder:**  Each Steering Council member can have the same funding organization as **at most one other Council member.**

The conda Organization [actively encourages](#provisional-members) participation by commercial entities.  However, we also need to preserve the voices of individual contributors, and prevent the conda Organization from ever favoring the financial interests of any commercial entity over the interests of the larger community.

The conda Organization enforces this restriction by tracking the funding sources of each Steering Council member.

### Shared Funding: Proximate, Ultimate, and In Between

> Generally, people have shared funding if they are an employee, officer, or member of the Board of Directors of the same entity; if they have a significant consulting relationship with the same entity; or if they own at least 1% of the equity or debt, or derivatives thereof, of the same entity.

Reviewer 2: Curious how we came to the 1% equity or debt provision. Think 5% is typically when one needs to make filings about involvement. Also debt doesn't usually give voting rights (except for some peculiar circumstances like restructuring, in which case it is converted to equity). Mentioning this as I'm wondering if this is unnecessarily restrictive on say entrepreneurs for example

Reviewer 3: Good point, @tnabtaf may be able to shed more light on where this came from.

Author: @jakirkham I copied this number straight from the PSF governance PEP.  That paragraph was written by Van Lindberg.  I don't have a strong opinion.

#### Tracking Funding

Nominated members will provide a list of organizations that fund above 25% of their time. Nominees will need to provide a complete “chain of funding” for each funding source. For example: “GitHub, a part of Microsoft”, or “CS Dept at Western Michigan University, *Computational Reproducibility Made Easy grant*, NHGRI, NIH, US Federal Government.”

The Steering Council determines the appropriate funding level(s) to consider when looking for common funding with existing and proposed members. The funding source of any Steering Council member can be put to a vote under the policy for [All Other Items](#all-other-items) below.

Council members need to keep their funding documentation up to date, and notify the council whenever it changes. Should a change in funding state create a shared funding situation, the council members should decide who among themselves should move to emeritus status. If no agreement can be reached, the Steering Council will hold a simple vote to determine which member is moved to emeritus status under the [Resolve Overlapping Funding](#resolve-overlapping-funding) voting provision below.

> In cases where people have absolutely no funding related to conda, we still document this funding state, but we do not require people to list their irrelevant funding. The "no funding" state can be held by an unlimited number of Steering Council members.

Reviewer 2: When does the "no funding" state apply? For example we could imagine a grad student or postdoc working on Conda in their free time. Do we count them as funded or not?

Reviewer 4: That is "no funding".

Reviewer 3: That's correct.

> What about cases where someone works on conda for work they are doing, but improving/directing the conda ecosystem is not a specific component of any of their funding? Generally, such cases should be considered as "no funding."

Reviewer 5: Am I right to interpret this that employees of a corporation who happen to work on the conda ecosystem but it isn't their direct job function should be considered "no funding"?

If so, is there anything stopping a malicious organization from having a bunch of people indirectly work and end up as "no funding" in order to seize control?

Reviewer 4: My understanding is that if someone pays you to work on conda, direct or indirect, then you are funded. If you only work on conda on unpaid time, then you are not funded. A grad student doing conda in their free time is thus unfunded, but folks at Voltron data working on conda as part of their job are funded.

Reviewer 2: Developers working on something on the weekend may be "not funded" in the same way as a grad student is. Think this is Keith's point.

If that work became important for the day job, that individual would then become funded. They would then be obligated to report their change in status per the governance document.

Since there is a limitation of 2 funded people per company, if there were others that were funded, the company would need to update who they want to keep in the council when this change occurred.

So no I don't think a company could takeover the steering council in this way, but they could mess up on process steps here and there could be consequences for those errors.

Reviewer 4: Right. These things can be put to a vote should something happen.

Reviewer 5: If we're trying to protect against a malicious entity then I think relying on people to report faithfully will not work. They could easily claim they're working on unpaid time and are therefore "not funded".

> My understanding is that if someone pays you to work on conda, direct or indirect, then you are funded. If you only work on conda on unpaid time, then you are not funded. A grad student doing conda in their free time is thus unfunded, but folks at Voltron data working on conda as part of their job are funded.

I understand you'd want to guard against a company like Voltron Data having too much influence, but this puts folks in a bit of a precarious situation. I.E. myself and others are primarily conda users and when we encounter sharp edges we try to help smoothen them out by contributing to the conda ecosystem. Voltron Data as a company doesn't track employee's time so there isn't really a concept of "company time" and because we work on so much OSS we generally know about everything going on with our employees OSS work. Our primary job functions and our paychecks are not based on our conda work, but Voltron Data is definitely happy to encourage and support OSS work.

Additionally, if someone values being on the conda steering committee it actively discourages them from considering companies that will possibly dictate that they need to move to emeritus status to keep their other 2 folks on the steering community, even if the job only leads to indirect contributions to the conda ecosystem.

Reviewer 4: If an entity goes through all of the trouble to convince a bunch of people to lie in an attempt to hijack the project, then I think there would be so many other repercussions that it'd render the conda organization as a group essentially non-functional anyways. 

@CJ-Wright has brought up similar points on employment.

Reviewer 4: FWIW, this vote will pass in ~6 hours depending on time zone which we did specify. 

Happy to revisit but it's pretty late in the hour on this one.

Reviewer 3: This is a good discussion and I appreciate your feedback on this @kkraus14, I look forward to deepening it in the future to make sure that unfunded individuals at companies feel represented in the community.

I want to echo what @jakirkham said about the steering council being a focal point for too many things and too little focus on OTHER ways to contribute, even though the governance policy has many ways to do that. E.g. project teams are perfectly reasonable ways to contribute to the conda ecosystem and are largely self-governed. For irregular contributors these are probably best suited.

The steering council, on the other hand, should really only deal with organizational problems, and administrative work and exists to make sure conda is governed sustainably.

IMHO the focus on the steering council alone (due to Anaconda dominating the project maintenance) was a reason why the governance policy (as designed originally 2 years ago) didn't result in more active participation. As such, the new steering council system that limits funding overreach exists to encourage more companies and organizations to put their money where their mouth is. That there is a part to that that is requiring some level of trust, should not surprise us given the Open Source nature of conda.

Reviewer 3: @beckermr Since it was not explicitly specified I assumed we're using the good old [Anywhere on Earth (AoE) "timezone"](https://en.wikipedia.org/wiki/Anywhere_on_Earth), to be as inclusive as possible. This has worked well in other Open Source communities that I've participated in the past.

Reviewer 4: I agree that this is the start, not the endpoint. Should we have the energy to pursue more governance modifications in the future, I'd be happy to help drive those along as well within reason.

Anywhere on earth is excellent and I will be using that a lot from now on!

Reviewer 5: > The point around recruitment of talent by companies being hindered by this rule is an interesting point (Idk if that particular concern came up)

It's actually the opposite I am more concerned about. If an individual is on the steering council and wishes to continue to be, it potentially hinders their ability to change companies and grow their career as other companies may already have 2 members on the steering council.

Reviewer 2: Yeah sorry Keith I may not have articulated that point well, but my intended meaning was the same as yours.

Author: @kkraus14: Yes. :-(

Weighted voting could help address this, while preserving limits on any one organization.

We need to find a cadence for making significant (requiring a vote) changes to governance.  We could reconsider weighting the next time we do this.

## Steering Council and Project Teams Interactions

> The Steering Council and the Project Teams have different roles and different motivations. conda governance aims to impose just enough oversight of Project Team procedures to ensure that membership and voting are indeed open and fair.

Reviewer 2: ```suggestion
The Steering Council and the Project Teams have different roles and different motivations. Conda governance aims to impose just enough oversight of Project Team procedures to ensure that membership and voting are indeed open and fair.
```

### Community Project Teams must be open to qualified new members

> Community projects must be open to new members who demonstrate their capability and interest in the project through working with the project as a contributor first. If an application to join a Community project is rejected, the person can appeal to the Steering Council. If the rejected person can prove that they in fact meet the criteria for becoming a Project Team member and that they were rejected for inappropriate reasons, including their funding source, then the rejection will be overturned. This voting item falls under the [Project Team Membership Appeal](#project-team-membership-appeal) policy below.

Reviewer 2: Would suggest reviewed vs. overturned simply because we may not be able to enumerate all possible issues that could occur and should keep the door open for maintaining the decision if that is needed for some reason

```suggestion
Community projects must be open to new members who demonstrate their capability and interest in the project through working with the project as a contributor first. If an application to join a Community project is rejected, the person can appeal to the Steering Council. If the rejected person can prove that they in fact meet the criteria for becoming a Project Team member and that they were rejected for inappropriate reasons, including their funding source, then the rejection will be reviewed. This voting item falls under the [Project Team Membership Appeal](#project-team-membership-appeal) policy below.
```

# Conda & Conda-Incubator Governance

This document outlines the policies and procedures that manage the conda community. This document recognizes that, while packaging and package tooling is deeply important to all of us, participation in the organization is voluntary. Individual members of the Steering Council, any sub-team, and external contributors choose to participate and may choose to leave at any time for any reason or for no reason.

Larger organizations, such as for-profit businesses, that contribute code and support to the conda organization are welcome to (but not required to) [maintain creative control](#federated-projects) of their contributions as long as such contributions adhere to the rules in this document. Such organizations are also welcome to [cede creative & maintenance control](#community-projects) of any specific project to the community as well.

We deeply appreciate all good faith contributions.

## Code of Conduct

The conda Organization's [Code of Conduct](CODE_OF_CONDUCT.md) governs all interactions within the conda Organization.  Please take a few minutes to familiarize yourself with it.

## Naming & Manifest

> This document defines the conda Organization as the body responsible for governing the tools, specifications, and libraries within the conda ecosystem, broadly speaking. This encompasses specific GitHub organizations, such as [`conda`](https://github.com/conda/) and [`conda-incubator`](https://github.com/conda-incubator/). Authors of particular software projects, such as [`conda`](https://github.com/conda/conda) or [`mamba`](https://github.com/mamba-org/mamba), are free to choose if and when their projects are included as part of the conda Organization.

Reviewer 6: ```suggestion
This document defines the conda Organization as the body responsible for governing the tools, specifications, and libraries within the conda ecosystem, broadly speaking. This encompasses specific GitHub organizations, such as [`conda`](https://github.com/conda/) and [`conda-incubator`](https://github.com/conda-incubator/). Authors of particular software projects, such as [`conda`](https://github.com/conda/conda) or [`mamba`](https://github.com/mamba-org/mamba), are free to choose when their projects are included as part of the conda Organization.
```

### The Rationale for Federated Projects

[Federated projects](#federated-projects) enable conda ecosystem projects to exist under the same umbrella organization, while maintaining community standards. They acknowledge the time and investment the developers and authors have put into their creations and that in certain cases they wish to maintain creative control. A federated system also allows for a phased transition plan for projects to become increasingly community maintained over time.

Federated projects cannot expect that the community will maintain or support the project as long as the project remains federated. If/when a federated project transitions to the community maintenance model, this project will be maintained and supported by the community.

Federated projects are under no obligation to have a transition plan or any intention of becoming a community maintained project. However, this is strongly encouraged and community maintained projects are seen as the ideal. Federated projects are seen as the reality.

> ## Trademarks

Author: @CJ-Wright is this a question about the conda Organization revoking trademarks?  I think that is covered by item 3.  But if you disagree, then item 3 could be expanded to explicitly mention that:

`The [Steering Council](https://github.com/conda-incubator/governance/blob/7cee49914f6b8082f0166ee0cef95712ab5e459c/README.md#steering-council) then makes trademark decisions, including the ability to revoke previously assigned trademarks, subject to agreed upon restrictions, and coordinates efforts with our fiscal sponsor.`

Would that address this?

> ## Trademarks

Author: @CJ-Wright: PR updated to explicitly include the ability to revoke trademarks that were previously granted.

[Community Projects](#community-projects) need to license/assign the project's trademarks when becoming a conda Organization Community Project. This is an act of good faith and communicates to the community that the project are committed to our overall efforts.

1. Trademarks are irrevocably *licensed or transferred* to the conda Organization's fiscal sponsor when a project is added to the conda Organization as a Community Project.
2. The conda Organization, in conjunction with our fiscal sponsor, then have the right to sublicense the trademarks to other projects, subject to specific limitations: The conda Organization may not degrade the distinctiveness of *licensed* trademarks, or use *licensed* trademarks to disparage or misrepresent the assigner. *Transferred* trademarks are completely controlled by the conda Organization and its fiscal sponsor.
3. The [Steering Council](#steering-council) then makes trademark decisions, subject to agreed upon restrictions, and coordinates efforts with our fiscal sponsor.

We have guidelines for making decisions about conda Organization trademarks:

> 1. The project should be open source with a permissive license

Author: We could include text [based on the description from Wikipedia](https://en.wikipedia.org/wiki/Permissive_software_license):

`1. The project should be open source with a permissive license.  A permissive license is a free-software license which instead of copyleft protections, carries only minimal restrictions on how the software can be used, modified, and redistributed.`

Would that be sufficient?

> 1. The project should be open source with a permissive license

Reviewer 4: @tnabtaf Does this exclude GPL or AGPL? I forget the definitions here. Excluding AGPL is fine by me but GPL should be allowed IMHO.

> 1. The project should be open source with a permissive license

Author: @CJ-Wright: The PR has been updated to explicitly include the definition of permissive that is used by OSI.

@beckermr: Several of us discussed GPL/copyleft at SciPy and none of us were eager to embrace the potential legal complications that come with copyleft.

## Steering Council Membership

Steering Council members need to 1) be invested in the conda ecosystem and community, and 2) have time and energy to contribute to conda Organization governance.  Council members, with the exception of [provisional members](#provisional-members), will have a demonstrated record of prior service to the conda community.

### Steering Council Size

The Steering Council should reflect a wide range of interests, and still be small enough to have a reasonable expectation that [quorum requirements](#quorum) can be met when votes are called.

The minimum and maximum sizes for the Steering Council are:

* Minimum: 9

> * Maximum: 21

Reviewer 7: I would add that if we have an upper limit, it would make sense to limit the number of participants from companies that are being paid to work for the company interest? so the council will not get controlled by any specific company and interests of them?

> * Maximum: 21

Author: @xhochy I added the upper limit after speaking with the IP lawyer Van Lindberg.  He argued that having an upper limit would encourage commercial participation (which is one of the goals of the update). If we don't have an upper limit then commercial entities will have no idea how dilute their influence can become. I hesitate to disagree with Van on anything...

We could bump this number up though, and still address the concern that Van raised.

> * Maximum: 21

Author: @marcelotrevisani Right now this proposal has hard limits 

* on the number of council members from the same company (2).  
* on the total number of council members that can be provisional members (33%).

Non-provisional members are members because they have a demonstrated history of contributing to the conda ecosystem.  You would never have a provisional member and a non-provisional member from the same company.

Does that address your concerns?

> * Maximum: 21

Reviewer 7: > @marcelotrevisani Right now this proposal has hard limits 
> 
> 
> 
> * on the number of council members from the same company (2).  
> 
> * on the total number of council members that can be provisional members (33%).
> 
> 
> 
> Non-provisional members are members because they have a demonstrated history of contributing to the conda ecosystem.  You would never have a provisional member and a non-provisional member from the same company.
> 
> 
> 
> Does that address your concerns?  

Yes, it does. Thank you! :)

> * Maximum: 21

Reviewer 4: I agree @tnabtaf. We already tolerate that risk and that seems fine.

> * Maximum: 21

Reviewer 4: @xhochy You ok here or do you have more questions?

> * Maximum: 21

Author: @xhochy: We dropped the maximum number of board members.  This helped us address several of the concerns that were raised by @CJ-Wright below.

> * Maximum: 21

Reviewer 8: Thanks, this makes me feel much better about the proposal!

The minimum of 9 works well with the [minimum quorum size of 5](#quorum) specified below. The maximum of 21 allows for a diversity of members, including as many as 7 Provisional Members (see below), and representation from 11 to 21 different participating organizations (see below).

> If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met. The requirement can be met by adding Provisional members, adding new contributing members, or members coming back from Emeritus status.

Author: > I think `all other Council business` is a bit vague. This most likely needs to call out explicitly what business can be performed, eg the election of new members.

Hmmm.  Every way I came up with to rewrite this statement:

`If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met.`

actually made things *less* clear.  The best I came up with was:

`If the Steering Council falls below its minimum, then the Steering Council's first priority is to meet this requirement.`

Which actually says something slightly different.  :-(

@CJ-Wright (or anyone), any suggestions would be most welcome.

> If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met. The requirement can be met by adding Provisional members, adding new contributing members, or members coming back from Emeritus status.

Author: > I'll also note that the quorum size would imply that elections of members to the Council would not be possible and so would effectively grind operations to a halt permanently.

I had not thought about the possibility of dropping below 5 Steering Council members.  That would require first dropping to 9 members, and then the mass simultaneous resignation of at least 5 members (a majority of the remaining council).

Unfortunately there are good reasons for that minimum quorum size of 5.  If we modify that requirement it would take a paragraph of text and more rules to explain an equivalent policy for minimum quorum size.

**@CJ-Wright, can I talk you into dealing with this situation if and when we get close to it?**  If we ever get close to just 9 members, then that implies that we need to reconsider a lot more than just minimum quorum size.  If we get close to 9 members then this whole approach needs to be revamped.

> If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met. The requirement can be met by adding Provisional members, adding new contributing members, or members coming back from Emeritus status.

Reviewer 4: We should not leave things to be resolved. They should be explicit and decided now. We have to let the minimum quorum rule be overrides by the vote timeout rules. If after a total of 3 emails, 3 chat messages, and a discussion at a community meeting over two weeks not more than five people respond to a vote then I think it should be fine for people to go ahead.

> If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met. The requirement can be met by adding Provisional members, adding new contributing members, or members coming back from Emeritus status.

Reviewer 4: In the topic of collaboration council business allowed if we do not meet membership requirements, clearly the only actions allowed should be votes related to changing the council membership. These are enumerated in the new doc already so we should simply list them.

> If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met. The requirement can be met by adding Provisional members, adding new contributing members, or members coming back from Emeritus status.

Reviewer 4: ```suggestion
If the Steering Council falls below its minimum, then all other Council business is suspended until this requirement is met. The only allowed council actions, in this case, are votes on adding Provisional members, votes on adding new contributing members, votes to resolve overlapping funding, existing members moving to Emeritus status, or existing members coming back from Emeritus status.
```

Reviewer 4: @CJ-Wright @tnabtaf @awwad please take a look here.

> If the Steering Council falls below its minimum, then all other Council business is suspended until this requirements is met. The requirement can be met by adding Provisional members, adding new contributing members, or members coming back from Emeritus status.

Author: @CJ-Wright We dropped the maximum number of council members from the proposal.  This will help address the concern about when someone leaves a company, but not the council.

> If the Steering Council membership exceeds its maximum, then all other Council business is suspended until this requirements is met. The requirement can be met by dropping Provisional members, removing members, or members voluntarily moving to Emeritus status.

Reviewer 4: ```suggestion
If the Steering Council membership exceeds its maximum, then all other Council business is suspended until this requirement is met. The only allowed council actions, in this case, are votes on removing Provisional members, votes on removing contributing members, votes to resolve overlapping funding, existing members moving to Emeritus status, or existing members coming back from Emeritus status.
```

Reviewer 4: @CJ-Wright @tnabtaf @awwad please take a look here.

No vote can be held that would push the Steering Council above its maximum size if the vote resulted in adding either standard or provisional Steering Council members.

### Provisional Members

As mentioned above, Steering Council members will usually have a demonstrated record of prior service to the conda community.  This requirement works well for ensuring that Council members are invested in the community before they can become part of community governance.

We also want to encourage participation by commercial and grant funded organizations that have committed to investing in the conda ecosystem. To encourage these organizations to commit resources to the conda ecosystem, **Provisional Steering Council Memberships** are available.

Provisional memberships are open to organizations that have committed resources to the conda Organization, but do not yet have an established record of contribution *within* the conda Ecosystem. Provisional memberships encourage investment, and give investing organizations an immediate voice in community decisions.

Provisional memberships are open to organizations that

* Are funding at least one FTE contributing to the conda ecosystem

> * Have not been involved in the conda ecosystem previously (and thus have no record of contributing to the ecosystem)

Author: Are there any objections if I merge this?

> * Have not been involved in the conda ecosystem previously (and thus have no record of contributing to the ecosystem)

Reviewer 3: This seems like a good suggestion that I've seen used elsewhere, I'm +1 on merging this.

> * Have not been involved in the conda ecosystem previously (and thus have no record of contributing to the ecosystem)

Reviewer 4: ditto for me on +1

These organizations can request 1 provisional membership on the Steering Council. The evaluation criteria for evaluating this request is based on the organization’s and the nominated person’s commitment to open source principles *as demonstrated in other projects.*

> Provisional memberships are expected to be temporary, lasting just long enough for contributors from that organization to demonstrate their values and earn representation as individual contributors.

Author: From @CJ-Wright

> If this is aimed at generating institutional buy in by organizations then the seat should sit with the organization.

And from @awwad 

> I think CJ is right that this needs some more language to make the provisional seat make sense as something an organization has.

Here I disagree.*  **I think that provisional seats are associated with both companies and individuals.  We want companies involved, but we want their representatives to be potential good community members**, not just whoever the company wants to put on the steering committee.

From @CJ-Wright 

> Should the Council be at maximum capacity that would lead to the disenfranchisement of an organization that otherwise invested in the ecosystem.

I think I am missing something here.  They would only be disenfranchised if the seat were filled while it is open.  Also I think this is a narrow window, as provisional members are only provisional until they can demonstrate that they are good contributors.  

Related to this is a discussion that happened before the vote was called, but that we didn't reach a clear decision on: Is 21 members a good maximum or should we raise that to a higher number?  Raising the maximum makes the disenfranchisement less likely (but still possible).

\* But I don't get a vote!

> Provisional memberships are expected to be temporary, lasting just long enough for contributors from that organization to demonstrate their values and earn representation as individual contributors.

Reviewer 3: We've talked about this in detail and would suggest to add the following to fix this:

> Organizations may submit an appeal to the Council to stay their provisional membership if and when the organization determines that the individual is no longer acting in the larger interests of the organization or conda community. The Council will consider the appeal and follow the same voting procedure as described in the "[Remove a Provisional Steering Council Member](#remove-a-provisional-steering-council-member)" section below. The organization may then nominate a replacement for the provisional membership following the procedure described in in the “[Nominate a Provisional Steering Council Member](#nominate-a-provisional-steering-council-member)” section below.

> Provisional memberships are expected to be temporary, lasting just long enough for contributors from that organization to demonstrate their values and earn representation as individual contributors.

Reviewer 3: We've talked about this in detail and would suggest to add the following to fix this:

> Organizations may submit an appeal to the Council to stay their provisional membership if and when the organization determines that the individual is no longer acting in the larger interests of the organization or conda community. The Council will consider the appeal and follow the same voting procedure as described in the "[Remove a Provisional Steering Council Member](#remove-a-provisional-steering-council-member)" section below. The organization may then nominate a replacement for the provisional membership following the procedure described in in the “[Nominate a Provisional Steering Council Member](#nominate-a-provisional-steering-council-member)” section below.

> Provisional memberships are expected to be temporary, lasting just long enough for contributors from that organization to demonstrate their values and earn representation as individual contributors.

Reviewer 4: SGTM!

> Provisional memberships are expected to be temporary, lasting just long enough for contributors from that organization to demonstrate their values and earn representation as individual contributors.

Reviewer 3: Weird that both comments made it, I blame the in-flight wifi :)

> Provisional memberships are expected to be temporary, lasting just long enough for contributors from that organization to demonstrate their values and earn representation as individual contributors.

Reviewer 3: I've made the change in cd9e3442a55c7df64fbd9130bc78dd3c118c1cbd.

Provisional memberships can also be revoked at any time by a simple majority vote of the Steering Council, if and when the Council determines that the organization or the individual is no longer acting in the larger interests of the conda community.

> Finally, **no more than 1/3 of the Steering Council membership can be Provisional Members.** At all times, at least 2/3 of the Steering Council membership hold a seat because of demonstrated contributions to the conda ecosystem. If this threshold is violated then all other Council business is suspended until this requirements is met. The requirement can be met by dropping Provisional members, or by adding contributing members.

Reviewer: See above discussion of "all other Council business"

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Author: Got it.  Will use whatever text we settle on above.

Author: This has been addressed.

> Finally, **no more than 1/3 of the Steering Council membership can be Provisional Members.** At all times, at least 2/3 of the Steering Council membership hold a seat because of demonstrated contributions to the conda ecosystem. If this threshold is violated then all other Council business is suspended until this requirements is met. The requirement can be met by dropping Provisional members, or by adding contributing members.

Reviewer 4: @CJ-Wright @tnabtaf @awwad please take a look here.

Meanwhile, across the world, Gouri works for GitHub, San-Yih works for LinkedIn, and João works on VS Code. Do they work for separate funders, or are they all funded by Microsoft?

We use this guideline for determining which level(s) of funding to use for membership eligibility:

> **When determining if funding is shared, choose the broadest level of funding where people might have a common set of goals because of their shared funding source.**

In Robert’s situation, two levels are good candidates for determining shared funding.

1. If Robert, or Robert’s department, is widely involved in many aspects of the conda ecosystem, then we may want to include his department in the list of Robert’s funders.
2. If Robert is working on a grant that spans many universities then we may want to include the grant.

In most circumstances we will not include the university as a funder because projects in the CS department, and in the Economics department, for example, are unlikely to have shared interests driven by both being at the same university. People funded by NSF, divisions of NSF, or the US government are also unlikely to have shared interests that are driven by their shared funding.

> In the other example, Microsoft is a large company and these three largely unrelated parts of it are unlikely to have common goals based mainly on their common ultimate funder. However, a company that wanted to control the conda Organization could do so by enlisting people from across their organization, claim separate funding, and then capture the council. **In this case, the default guideline is to treat all parts of a company as having one funder.**

Reviewer: How is "a company" determined here? Many companies have distinct legal entities. How does this handle investors? Eg. could a VC firm corner the market on conda by strategically investing in certain firms?

Do we expect members of the Council to reject grants or employment because it may cause them to no longer have voting rights? Asking maintainers to choose between their livelihood and the Conda project seems punitive and counter to a volunteer based organization.

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Reviewer 9: This does feel like a problem, but no simple solution immediately occurs to me.

Author: @CJ-Wright and @awwad: This guideline is suggested in the document:

> We use this guideline for determining which level(s) of funding to use for membership eligibility:
>
>> When determining if funding is shared, choose the broadest level of funding where people might have a common set of goals because of their shared funding source.

There is also a long discussion on proximate and ultimate funding, including long chains of funding.  So, I think the text covers the VC situation.  But the two of you don't think it does.  **What text could we add to cover this situation?**

Author: From @CJ-Wright 

> Do we expect members of the Council to reject grants or employment because it may cause them to no longer have voting rights? 

**The short answer is YES. This is exactly the tradeoff people will have to make.**  

If they switch to a funding source that already has 2 people on the Steering council, then one of those 3 people will have to lose their voting position.  That third person can still have a voice, just not a vote. For example, I don't have a vote, but as this PR shows, I still have a voice here.

This proposal seeks to strike a balance between encouraging involvement, preventing capture, and preferring simpler solutions over more complex ones.  (We did scope out a fractional voting system where each additional council member with the same funding would result in a fractional increase in the voting pool.  It would have partially addressed your concern, but it would have added a lot of complication to voting.)

This balance requires tradeoffs.  We could have different tradeoffs, but I don't see a solution that avoids them and still achieves the stated goals.

Open to suggestions (and I can probably find the fractional voting proposal if you want to see it).

Reviewer 9: If folks are okay with the steering council voting to decide case by case on complex funding situations, then that makes sense enough to me.  Yes, the arrangement is a bit punitive, but Dave is right that there's a high complexity tradeoff to contend with.

Author: @CJ-Wright: We have added legal text ripped straight from the Python PEP about this.  That text was written by Van Lindberg when Python was dealing with this.

Reviewer 3: This was NOT from a Python PEP but from the PSF by-laws: https://github.com/python/psf-bylaws/commit/4459f6eeee40f846540dccd655996b66365d6cd7.

#### Tracking Funding

Nominated members will provide a list of organizations that fund above 25% of their time. Nominees will need to provide a complete “chain of funding” for each funding source. For example: “GitHub, a part of Microsoft”, or “CS Dept at Western Michigan University, *Computational Reproducibility Made Easy grant*, NHGRI, NIH, US Federal Government.”

The Steering Council determines the appropriate funding level(s) to consider when looking for common funding with existing and proposed members. The funding source of any Steering Council member can be put to a vote under the policy for [All Other Items](#all-other-items) below.

> Council members need to keep their funding documentation up to date, and notify the council whenever it changes. Should a change in funding state create a shared funding situation, the council members should decide who among themselves should move to emeritus status. If no agreement can be reached, the Steering Council will hold a simple vote to determine which member is moved to emeritus status under the [Resolve Overlapping Funding](#resolve-overlapping-funding) voting provision below.

Reviewer: What is the enforcement mechanism for failure to comply or keep records up to date? What is the time period for funding updates?

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Author: The council always has the power to kick people off the council.  We could state that here, but it is covered elsewhere.

I am not opposed to restating that here.

I actually lean against specifying a time line here.  We could say something like "Within two weeks." Or, in practice, we could say "Hey! Update your status!"  If someone does not, remind them in two weeks.  If they still don't then find out what is going on.  Do we need to codify that?

Also see my general comment in response to your general comment.

Author: @CJ-Wright: The text has been updated with a specific deadline for updating funding information, and specified action to take if that deadline is missed.

> In cases where people have absolutely no funding related to conda, we still document this funding state, but we do not require people to list their (irrelevant) funding. The "no funding" state can be held by an unlimited number of Steering Council members.

Reviewer: How is this "no funding" determined? If someone works on Conda to patch bugs found in the pursuit of a particular grant were they funded to work on Conda?

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Author: @CJ-Wright It's clear to me (in my brain!) that the answer to this specific question is "No," the person is not funded to work on conda.  However, my brain is not part of this document.

What about:

`In cases where people have absolutely no funding related to conda, we still document this funding state, but we do not require people to list their (irrelevant) funding. The "no funding" state can be held by an unlimited number of Steering Council members.`

`What about cases where someone works on conda for work they are doing, but improving/directing the conda ecosystem is not a specific component of any of their funding?  Generally, such cases should be considered as "no funding."`

`(And then we will thank these people! Volunteers are a big part of the conda community and many components of the ecosystem would not exist without their contributions. The conda community absolutely wants these contributors involved in the community.`

Ok, that gets a little preachy towards the end. We could not include that last paragraph.

Reviewer 9: I think that `improving/directing the conda ecosystem is not a specific component of any of their funding` can reasonably be argued to cover most folks.

Author: @CJ-Wright & @awwad: More descriptive text has been added.

### Quorum

With one exception, all percentages below express *both* required participation, as a fraction of active Steering Council members, as well as the fraction of those who vote affirmatively (i.e. vote `yes`) on the issue. The fraction of those who voted affirmatively is determined from the total number who voted `yes`, `no` or `abstain`, not the size of the active Steering Council. The exception is when 50% is required. In these cases, *at least half* of team members need to participate, but *more than 50%* of votes need to be affirmative for the vote to pass. Some votes have special quorum rules and those are listed explicitly below.

For example, in a vote requiring 50% participation, with 20 active members, at least 10 must vote; if 10 or 11 vote, there must be at least 6 affirmative votes. If 12 or 13 members vote, at least 7 votes must be affirmative to pass, and so on.

> To prevent any [funding organization](#shared-funding-membership-restrictions) from swaying the majority of votes on any topic, the minimum quorum for a vote must be at least 5 members, or the specified council membership percentage for this type of vote, whichever is greater.

Reviewer 4: ```suggestion
To prevent any [funding organization](#shared-funding-membership-restrictions) from swaying the majority of votes on any topic, the minimum quorum for a vote must be at least 5 members, or the specified council membership percentage for this type of vote, whichever is greater. If the council membership falls below 5 members, then the minimum quorum size is set to the size of the council regardless of the vote percentage.
```

Reviewer 4: @CJ-Wright @tnabtaf @awwad please take a look here.

Because everyone is busy and it is difficult sometimes to get quorum, votes not achieving quorum will eventually time out on their set end date. When this happens, the current participation level is taken for what it is, and the percentage of affirmative votes is calculated from whatever the vote total is at that time. In order for a timeout to occur, the vote must have:

> * been open for at least 2 weeks

Reviewer 4: ```suggestion
* met the minimum quorum described above
* been open for at least 2 weeks
```

  * at least 3 members voting, with all votes being "yes"
  * at least 50% participation, with greater than 50% "yes" votes to pass

> ## Ratification of this Document

Reviewer 2: Did we intend to remove this section?
